packages/debugger-extension/schema/main.json

```diff
@@ -30,6 +30,10 @@
       {
         "command": "debugger:render-mime-variable",
         "selector": ".jp-DebuggerVariables-body"
+      },
+      {
+        "command": "debugger:copy-to-clipboard",
+        "selector": ".jp-DebuggerVariables-body li:not(.jp-DebuggerVariables-noCopy)"
```

Comment:
Is it better to hide the menu item if there is no value to copy, or to show it greyed out (disabled)? I think it depends on the size of the context menu (if there are many items hide, if few show but disable).

Reply:
I don't think we can disable a menu entries using the plugin settings.
The command itself does not receive information about the DOM element that renders it.
I don't know if there is a way to get such information from the `isEnabled` option of the command.

Reply:
> The command itself does not receive information about the DOM element that renders it.

True (which is simultaneously a good and a bad thing), but `isEnabled()` can do perform hit testing using `contextMenuHitTest` to find the relevant node.

Reply:
To clarify, I am not insisting that this is the way to go, just brainstorming.

Reply:
Nice, I didn't this function, that's helpful, thanks.

> To clarify, I am not insisting that this is the way to go, just brainstorming.

Currently there is only 2 or 3 entries (respectively tree or grid view) so I agree it's better to disable it.
I did it in https://github.com/jupyterlab/jupyterlab/pull/13670/commits/6b40b6f46e1c5e4dea277185c7896ed09bbe7e45.

The way it was done before was not working for grid view, now the entry is always available, no matter the variable (only in grid view).

Reply:
Well, `DataGrid` has a `hitTest` method too. It may require modifying the mouse handler but it should be rather straightforward as we already do this for the debugger.

Reply:
Finally there was an simpler way to do it for both tree view and grid view: checking the `selectedVariable.value` to enable the menu entry or not https://github.com/jupyterlab/jupyterlab/pull/13670/commits/3b56b256bb24538e72490b8794a793b5ec9a2705

I also removed the class `jp-DebuggerVariables-emptyValue` which is now useless, and added a test on the disable entry.